cpp/modules/deck.gl/layers/src/solid-polygon-layer/solid-polygon-layer.h

```diff
@@ -94,11 +94,11 @@ class SolidPolygonLayer::Props : public Layer::Props {
 };
 
 /// The order of fields in this structure is crucial for it to be mapped to its GLSL counterpart properly.
-/// bool has a 4-byte alignment in GLSL.
+/// Using uint32_t in place of bool in order to avoid packing issues, even when specifying 4-byte alignment.
 /// https://learnopengl.com/Advanced-OpenGL/Advanced-GLSL
 struct SolidPolygonLayerUniforms {
-  alignas(4) bool extruded;
-  alignas(4) bool wireframe;
+  uint32_t extruded;
```

Comment:
Do we actually know what the size of `bool` was when it was failing? I would be concerned that the size would change unexpectedly between compilers. Using `uint32_t` is certainly safer.

Reply:
The size didn't vary, it was packing that made the memory be structured differently. Even with `#pragma pack` directives, I couldn't get it to pack these fields into 4-byte chunks with added padding consistently.

Perhaps I could've made it work with a bit more effort, but without access to all the compilers/versions that this could be compiled with, it's hard to say whether it's a fail-proof implementation like this one is supposed to be